### PR TITLE
feat(notification): add experimental notification 

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/IBM/carbon-components/issues/1295

Adds in experimental `Notification`

#### Changelog

**New**

- Experimental `Notification`
- A few new color variables
- ~`@mixin exp-breakpoint($size)`, which utilizes the new breakpoint scale~ Added to existing breakpoint mixin

**Changed**

- Changed the `notification.hbs` file to conditionally render icons on toast components
- Added long text to test long strings (will remove before merge)

**Removed**

- Extra styles

#### Testing / Reviewing

[Staging Link](http://carbon-dev-environment-exp-notification-active-lemur.mybluemix.net/?nav=notification)

screen shot 2018-10-30 at 3 40 11 pm

### CODE REVIEW

File: src/globals/scss/_layout.scss [hunk 1]
```diff
@@ -4,12 +4,20 @@ $breakpoints: (
   bp--xs--major: 500px,
   bp--sm--major: 768px,
   bp--md--major: 992px,
-  bp--lg--major: 1200px
+  bp--lg--major: 1200px,
+);
+
+$exp-breakpoints: (
+  sm: 320px,
```
Comment (asudoh) [inline src/globals/scss/_layout.scss:8]: Given change in variable name tends to a breaking change, 🤔 of a way not having to change this variable name before shipping the code in production. Wondering if one of the following would make sense:

1. (Given there no conflict in item names) Merging new breakpoints to `$breakpoints`, only in experimental mode
2. Figure out a name that we don't have to change before shipping the code in production

**Review by asudoh — APPROVED** [review]
LGTM basically 👍, just a thought for (near) future - Thanks @tw15egan!

File: src/globals/scss/_layout.scss (hunk 1 above)
Comment (tw15egan) [reply src/globals/scss/_layout.scss:8]: That's a good point. Perhaps it would make the most sense to just add them to the existing map regardless.

Comment (alisonjoseph) [reply src/globals/scss/_layout.scss:8]: double checking with @joshblack but I believe the updated breakpoints should be pulled in from the layouts package in carbon-elements

https://github.com/IBM/carbon-elements/blob/master/packages/layout/scss/_breakpoint.scss

**Review by alisonjoseph — APPROVED** [review]
These look great! Just a question on how we want to handle the breakpoints for now.

File: src/globals/scss/_layout.scss (hunk 1 above)
Comment (asudoh) [reply src/globals/scss/_layout.scss:8]: @tw15egan Thank you for sharing your thoughts! Just merging the two will make sense if we see it's too early to incorporate `carbon-elements` (thank you for bringing this up @alisonjoseph!) and we don't see the breakpoints will change in future (meaning, the list of breakpoints is not seen as experimental). Using its [`carbon-elements`' `_breakpoint.scss`](https://github.com/IBM/carbon-elements/blob/v0.0.1-alpha.2/packages/layout/scss/_breakpoint.scss) is fine from my perspective from what I see the code, while we need to evaluate `carbon-elements` incorporation case-by-case basis given its nature of new fundamental codebase.

Comment (tw15egan) [reply src/globals/scss/_layout.scss:8]: For now, I've just added them to the existing breakpoint mixin. When we start using Carbon Elements, we will need to update all components

Comment (asudoh) [reply src/globals/scss/_layout.scss:8]: Makes sense - Thanks @tw15egan for your quick update!

**Comment by tw15egan** [comment]
@jeanservaas changes have been made: http://carbon-dev-environment-exp-notification-active-lemur.mybluemix.net/?nav=notification

**Comment by tw15egan** [comment]
@jeanservaas has approved this irl

**Pull request merged**